clickhouse/columns/array.cpp

```diff
@@ -5,43 +5,58 @@
 namespace clickhouse {
 
 ColumnArray::ColumnArray(ColumnRef data)
+    : Column(Type::CreateArray(data->Type()))
+    , data_(data->CloneEmpty())
```

Comment:
Removed that, since it breaks previous user code and complicates implementation of other methods. Just added a warning comment that users shouldn't abuse this constructor.